### PR TITLE
Add "button" class to the QR code link

### PR DESCRIPTION
**Pull request by Jerome-Herbinet**
## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/settings/src/components/AuthTokenSetupDialogue.vue [hunk 1]
```diff
@@ -70,7 +70,7 @@
 		</div>
 		<div class="app-password-row">
 			<span class="app-password-label" />
-			<a v-if="!showQR"
+			<a class="button" v-if="!showQR"
```
Comment (szaimen) [inline apps/settings/src/components/AuthTokenSetupDialogue.vue:73]:
```suggestion
			<a class="button"
			v-if="!showQR"
```

File: apps/settings/src/components/AuthTokenSetupDialogue.vue [hunk 2]
```diff
@@ -70,7 +70,8 @@
 		</div>
 		<div class="app-password-row">
 			<span class="app-password-label" />
-			<a class="button" v-if="!showQR"
+			<a class="button"
+			v-if="!showQR"
```
Comment (szaimen) [inline apps/settings/src/components/AuthTokenSetupDialogue.vue:74]:
```suggestion
				v-if="!showQR"
```

**Comment by Jerome-Herbinet** [comment]
@szaimen I let you check if everything is OK for you, following your 2 co-author commits.

**Review by artonge — COMMENTED** [review]
Should we take the opportunity to use `NcButton` instead?
Using a `<button>` should also work.

In any case, keeping an `<a>` is semantically wrong.

**Comment by szaimen** [comment]
> Should we take the opportunity to use `NcButton` instead?

sounds good!

**Comment by Jerome-Herbinet** [comment]
@artonge , will simply be replacing : 

```
<a 
		v-if="!showQR"
		@click="showQR = true">
		{{ t('settings', 'Show QR code for mobile apps') }}
</a>
```

with :

```
<button 
		v-if="!showQR"
		@click="showQR = true">
		{{ t('settings', 'Show QR code for mobile apps') }}
</button>
```

... do the trick ? In other words, will the behaviour that makes appear the QR code still work ?

**Comment by artonge** [comment]
@Jerome-Herbinet from reading the code, I would say yes, but a test would be welcome.
The optimal solution would be to use `NcButton`.

**Comment by Jerome-Herbinet** [comment]
> @Jerome-Herbinet from reading the code, I would say yes, but a test would be welcome :). The optimal solution would be to use `NcButton`.

@artonge can you give me details about `NcButton` usage method ?

File: apps/settings/src/components/AuthTokenSetupDialogue.vue [hunk 3]
```diff
@@ -70,7 +70,8 @@
 		</div>
 		<div class="app-password-row">
 			<span class="app-password-label" />
-			<a v-if="!showQR"
+			<a class="button"
+				v-if="!showQR"
 				@click="showQR = true">
 				{{ t('settings', 'Show QR code for mobile apps') }}
 			</a>
```
Comment (artonge) [inline apps/settings/src/components/AuthTokenSetupDialogue.vue:77]:
```suggestion
			<NcButton v-if="!showQR"
				@click="showQR = true">
				{{ t('settings', 'Show QR code for mobile apps') }}
			</NcButton>
```

**Review by artonge — COMMENTED** [review]
It is used a few lines above.
Here is the documentation: https://nextcloud-vue-components.netlify.app/#/Components/NcButton

**Comment by Jerome-Herbinet** [comment]
> It is used a few lines above. Here is the documentation: https://nextcloud-vue-components.netlify.app/#/Components/NcButton

OK, thanks @artonge  ; didn't see it.

**Comment by Jerome-Herbinet** [comment]
Closed, due to conflict I don't have time to try to solve.
New PR : https://github.com/nextcloud/server/pull/39015

**Pull request closed without merging**